client/my-sites/upgrades/domain-management/transfer/transfer-to-other-site/index.jsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+	get,
+	omit,
+} from 'lodash';
+import page from 'page';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import Card from 'components/card';
+import SiteSelector from 'components/site-selector';
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getSites } from 'state/selectors';
+import Header from 'my-sites/upgrades/domain-management/components/header';
+import Main from 'components/main';
+import paths from 'my-sites/upgrades/paths';
+import { getSelectedDomain } from 'lib/domains';
+import NonOwnerCard from 'my-sites/upgrades/domain-management/components/domain/non-owner-card';
+import DomainMainPlaceholder from 'my-sites/upgrades/domain-management/components/domain/main-placeholder';
+import SectionHeader from 'components/section-header';
+import TransferConfirmationDialog from './confirmation-dialog';
+import { successNotice, errorNotice } from 'state/notices/actions';
+import wp from 'lib/wp';
+
+const wpcom = wp.undocumented();
+
+class TransferToOtherSite extends React.Component {
+
+	static propTypes = {
+		selectedDomainName: React.PropTypes.string.isRequired,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
+		currentUser: React.PropTypes.object.isRequired
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			targetSiteSlug: '',
+			showConfirmationDialog: false,
+			disableDialogButtons: false
+		};
+	}
+
+	isDataReady() {
+		return this.props.domains.hasLoadedFromServer;
+	}
+
+	checkSiteEligibility = ( site ) => {
```

Comment:
`checkSiteEligibility` seemed off to me too, but I didn't realize why. Thinking about it as a predicate makes total sense 👍